tsdb/engine/tsm1/reader.go

```diff
@@ -607,7 +607,7 @@ func (d *indirectIndex) UnmarshalBinary(b []byte) error {
 		i++
 
 		// 2 byte count of index entries
-		count := btou16(b[i : i+indexCountSize])
+		count := int32(btou16(b[i : i+indexCountSize]))
```

Comment:
You can't see where it is incremented in the diff.  It happens on here: https://github.com/influxdb/influxdb/pull/5264/files#diff-2226d419783ec10497f3bab10d1c4300R619

See this snippet for essentially what is happening: https://play.golang.org/p/zMYh_UOw4h